src/bin/cli.js

```diff
@@ -6,138 +6,155 @@ const chalk = require('chalk');
-          if (err) {
-            console.log(`\nError in accessing provided ${chalk.blueBright('scenario')} file \nDetails:${err}\n\n`);
-            inputLoopScenario(handlingContext);
-          } else if (!String(handlingContext.scenarioFile).match(handlingContext.yamlJsonRegex)) {
-            console.log(`\nPlease provide a proper ${chalk.blueBright('scenario')} filepath ex: ./myAsyncApi.json ./myAsyncApi.yaml:`);
-            inputLoopScenario(handlingContext);
+const inputLoopScenario = (rd,scenario,regex) => {
+  const res = checkFilepath(scenario,regex);
+
+  if (!res) {
+    return new Promise((resolve) => {
+      rd.question('\nPlease provide an existent yaml or json file .It should abide by the scenario json schema.\nScenario filepath:',(answer) => {
+        const inputLoop = (filepath) => {
+          const res = checkFilepath(filepath,regex);
+          if (!res) {
+            rd.question('Please fix errors and provide a correctly formatted and accessible file in filepath.\nScenario filepath:',(answer) => {
+              inputLoop(answer);
+            });
           } else {
-            //handlingContext.scenarioFile = handlingContext.scenarioFile;
-            handlingContext.scenarioReady = true;
-            parseAsyncApi(handlingContext);
+            resolve(filepath);
           }
-        }
-        );
-      }
+        };
+
+        inputLoop(answer);
+      });
     });
+  }
+  return  new Promise((resolve) => {resolve(scenario);});
+};
+
+const inputLoopAsyncApi = (rd,asyncFile,regex) => {
+  const res = checkFilepath(asyncFile,regex);
+
+  if (!res) {
+    return new Promise((resolve) => {
+      rd.question('\nPlease provide an existent yaml or json file.It should abide by the asyncApi Spec.\nAsyncApi filepath:',(answer) => {
+        const inputLoop = (filepath) => {
+          const res = checkFilepath(filepath,regex);
+          if (!res) {
+            rd.question('Please fix errors and provide a correctly formatted and accessible file in filepath.\nAsyncApi Filepath:',(answer) => {
+              inputLoop(answer);
+            });
+          } else resolve(filepath);
+        };
+
+        inputLoop(answer);
+      });
+    });
+  }
+  return new Promise((resolve) => {
+    resolve(asyncFile);
   });
 };
 
 /**
- * Verifies the command line arguments, re-prompts in case of error. Parses file and returns the object representation.
+ * Verifies the command line arguments, re-prompts in case of error. Parses asyncApiF and returns the object representation.
  * @returns {Promise<*|null>}
  * @param rd
- * @param file
+ * @param asyncApiF
  * @param scenarioFile
  */
-const verifyInput_ParseAndLinkFiles =  async (rd, file,scenarioFile) => {
-  const handlingContext = this;
-  handlingContext.ready = false;
-  handlingContext.rd = rd;
-  handlingContext.file = file;
-  handlingContext.scenarioFile= scenarioFile;
+const verifyInput_getData =  async (rd, asyncApiF,scenarioFile) => {
```

Comment:
I must say I am not a huge fan of the naming of the functions, especially in terms of `_`, generally the common consensus is using camelCase for all function namings to keep it consistent. Do you think we should rename it?

Reply:
ok, you dont like the naming of some other function?